crates/iota-indexer/src/ingestion/snapshot/orchestration.rs

```diff
@@ -1,50 +1,220 @@
 // Copyright (c) 2025 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use iota_data_ingestion_core::WorkerPool;
+use std::collections::HashMap;
+
+use anyhow::Context;
+use iota_data_ingestion_core::{IndexerExecutor, ReaderOptions, WorkerPool};
 use iota_metrics::get_metrics;
+use tokio::{
+    task::JoinHandle,
+    time::{Duration, sleep},
+};
 use tracing::info;
 
 use crate::{
-    PgIndexerStore,
+    CancellationToken, PgIndexerStore,
     config::SnapshotLagConfig,
     ingestion::{
+        common::{
+            orchestration::{ShimIndexerProgressStore, new_executor},
+            persist::Writer,
+        },
         primary::persist::TransactionObjectChangesToCommit,
         snapshot::{persist::ObjectSnapshotWriter, prepare::ObjectsSnapshotWorker},
     },
     metrics::IndexerMetrics,
+    spawn_monitored_task,
     types::IndexerResult,
 };
 
 const OBJECT_SNAPSHOT_CHANNEL_CAPACITY: usize = 600;
+const WAIT_FOR_SNAPSHOTTABLE_DATA_POLL_INTERVAL: Duration = Duration::from_secs(1);
```

Comment:
Nit: I thought I had added a comment but apparently I hadn't.

I would like to suggest to rename `wait_for_snapshottable_data` into something that is more proper English and still maintain the meaning. `wait_for_snapshot_lag` could be an idea.

Reply:
done